From 1aa95676919986dcbc2d5677760e66832173020f Mon Sep 17 00:00:00 2001
From: Jean-Christophe Trotin <jean-christophe.trotin@st.com>
Date: Wed, 7 Sep 2016 12:00:51 +0200
Subject: [PATCH 11/23] v4l2object: force capture-io-mode if "memory:DMABuf" in
 caps

capture-io-mode is forced to GST_V4L2_IO_DMABUF only if "memory:DMABuf"
feature is set in the src caps.

Change-Id: I35661b68396a4edc7e867ffff72e04d09b9d669d
Reviewed-on: https://gerrit.st.com/57584
Reviewed-by: Jean Christophe TROTIN <jean-christophe.trotin@st.com>
Tested-by: Jean Christophe TROTIN <jean-christophe.trotin@st.com>
Reviewed-by: Benjamin GAIGNARD <benjamin.gaignard@st.com>
Reviewed-by: Vincent ABRIOU <vincent.abriou@st.com>
---
 sys/v4l2/gstv4l2object.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 8bd8b07..c9649e5 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -2778,6 +2778,8 @@ gst_v4l2_object_setup_pool (GstV4l2Object * v4l2object, GstCaps * caps)
   GST_V4L2_CHECK_NOT_ACTIVE (v4l2object);
 
   if (!V4L2_TYPE_IS_OUTPUT (v4l2object->type) &&
+      gst_caps_features_contains (gst_caps_get_features (caps, 0),
+          GST_CAPS_FEATURE_MEMORY_DMABUF) &&
       v4l2object->req_mode != GST_V4L2_IO_DMABUF) {
     GST_WARNING_OBJECT (v4l2object->element,
         "Forcing capture-io-mode to GST_V4L2_IO_DMABUF by default...");
-- 
2.7.4

